### ISSUE TITLE
False positive about @UtilityClass and MemberName

### ISSUE THREAD

**Issue by LynnBroe** [checkstyle/checkstyle#12963]
I have read check documentation: https://checkstyle.org/config_naming.html#MemberName
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac A.java


/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="SuppressWarningsFilter"/>
  <property name="charset" value="UTF-8"/>
  <property name="severity" value="warning"/>
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">
    <module name="MemberName">
      <property name="format" value="^[a-z][a-z0-9][a-zA-Z0-9]*$"/>
      <message key="name.invalidPattern"
               value="Member name ''{0}'' must match pattern ''{1}''."/>
    </module>
  </module>
</module>


/var/tmp $ cat A.java
import lombok.experimental.UtilityClass;
@UtilityClass
class A {
    private String fFirstName;
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.9.3-all.jar -c config.xml A.java
Starting audit...
[WARN] D:\MVN\newtest\src\main\java\.\A.java:6:20: Member name 'fFirstName' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$'. [MemberName]
Audit done.

```
However, the annotation @UtilityClass annotation from Lombok can transform the source code into the following code example:
```
final class A {
    private static String fFirstName;
    
    private A() {
        throw new java.lang.UnsupportedOperationException("This is a utility class and cannot be instantiated");
    }
}
```
```
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.9.3-all.jar -c config.xml gen/A.java
Starting audit...
Audit done.
```

---

**Describe what you expect in detail.**

---
The `A.java` and `gen\A.java` are the same. The `@UtilityClass` annotation will make the member field `static`, so the analysis results are not the same. Hence, I think it should be a false positive and the rule should consider the analysis results.

**Comment by romani** [comment]
ignoreStatic is true by default.
https://checkstyle.org/writingchecks.html#Limitations

Checkstyle is not aware of generated version of code.

You need to use some suppression, for example https://checkstyle.org/config_filters.html#SuppressionXpathSingleFilter and provide xpath that suppress all violations from this Check on classes with such annotation.

Please use suppression a lot as you use Lombok on classes, checkstyle are not aware of code generation, there are a lot of code generation tools in world we cannot support all, please wisely use suppression/filter. We have a lot of them and they are very flexible.

**Comment by romani** [comment]
workaround is to use suppression
see example at https://github.com/checkstyle/checkstyle/issues/12962#issuecomment-3127427086 

```
    <module name="SuppressionXpathSingleFilter">
      <property name="checks" value="AbbreviationAsWordInName"/>
      <property name="query" value="//CLASS_DEF[.//ANNOTATION/IDENT[@text='UtilityClass']]/OBJBLOCK/VARIABLE_DEF/IDENT"/>
    </module>
```
